source/common/http/filter_manager.cc

```diff
@@ -1335,7 +1335,9 @@ void FilterManager::setBufferLimit(uint32_t new_limit) {
 }
 
 void FilterManager::contextOnContinue(ScopeTrackedObjectStack& tracked_object_stack) {
-  tracked_object_stack.add(connection_);
+  if (connection_) {
```

Comment:
OOC, is this a bug fix, or an artifact of adding upstream filters?

Reply:
Oh, I guess it's because connection_ is now a pointer instead of a ref and it's a pointer that's allowed to be null? I think the Envoy idiom would be OptRef<Network::Connection> instead of a nullable pointer. Would that work?

All of THAT having been said, why is connection_ allowed to be null now? (This is probably obvious)

Reply:
so we can have filters on async client requests (no downstream)